syntax = "proto3";

package booksuggestion;

service BookSuggestionModelService {
  rpc GetBookRecommendations(RecommendationsRequest) returns (BookRecommendations);
}

message RecommendationsRequest {
  string order_json = 1;
}

message Book {
  string id = 1;
  string title = 2;
  string author = 3;
  string description = 4;
  int32 copies = 5;
  int32 copiesAvailable = 6;
  string category = 7;
  string img = 8;
  int32 price = 9;
}

message BookRecommendations {
  repeated Book books = 1;
}
